core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java

```diff
@@ -51,13 +53,11 @@
     private LogLevel failedResponseLogLevel = LogLevel.WARN;
     private Function<? super RequestOnlyLog, LogLevel> requestLogLevelMapper =
             log -> requestLogLevel();
-    private Function<? super RequestLog, LogLevel> responseLogLevelMapper =
+    private ResponseLogLevelMapper responseLogLevelMapper =
```

Comment:
I find the order of composition somewhat confusing because the lastly added one takes precedence:

```
// bar takes precedence
builder
  .responseLogLevelMapper(foo)
  .responseLogLevelMapper(bar)
```

How about calling `this.responseLogLevelMapper.orElse(newResponseLogLevelMapper)` in `responseLogLevelMapper()` and applying the fallback mapper (`log -> log.responseCause() == null ? ...`) at the end in `build()`, so that it is like `foo.orElse(bar).orElse(fallbackMapper)`?

Reply:
Thanks!! More natural.